modules/sort/main/index.coffee

```diff
@@ -1,15 +1,17 @@
 # Intl.Collator isn't supported by safari
 # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Collator
 # using a collator is supposed to be faster than doing localeCompare
-hasCollator = Intl?.Collator?
+hasCollator = -> Intl?.Collator?
 
-collator = if hasCollator
+collatorFn = -> if hasCollator()
   new Intl.Collator(undefined, {numeric: true}).compare
 else
   (a, b) ->
-    if a < b then -1
-    else if a > b then 1
-    else 0
+    if a is b then 0
+    else if String(a) > String(b) then 1
```

Comment:
This makes the behaviour of the custom collator more consistent with the behaviour of Intl.Collator, since it also turns `undefined` into 'undefined' and `null` into 'null'.